cpp/src/gandiva/precompiled/time.cc

```diff
@@ -944,6 +944,99 @@ gdv_int64 castBIGINT_daytimeinterval(gdv_day_time_interval in) {
          extractDay_daytimeinterval(in) * MILLIS_IN_DAY;
 }
 
+static const char* PATTERN_MONTH[] = {"MONTH", "MON", "MM"};
+static const int PARTTERN_MONTH_LEN[] = {5, 3, 2};
+
+static const char* PATTERN_YEAR[] = {"YEAR", "YYYY", "YY"};
+static const int PARTTERN_YEAR_LEN[] = {4, 4, 2};
+static const char* MAXDATE = "9999-12-31 23:59:59";
+
+FORCE_INLINE
+const char* date_trunc_timestamp_utf8(int64_t context, int64_t date,
```

Comment:
If it does not already exist a function that receives a timestamp as a parameter, I think you can create. You should only make a call to this function.